tests/js/client/shell/shell-debug-dump.js

```diff
@@ -0,0 +1,345 @@
+      explainer.debugDump(fileName, query, {}, {});
+      const outFileName = fs.getTempFile() + "-inspectDump";
+      explainer.inspectDump(fileName, outFileName);
+      recreateEmptyDatabase();
+      internal.load(outFileName);
+      const graphAfter = db._graphs.document(graphName);
+      delete graphAfter._rev;
+      assertEqual(graphBefore, graphAfter, "graph recreated after debugDump is not the same as the original");
+      assertNull(db[vn1].any());
+      assertNull(db[vn2].any());
+      assertNull(db[cn1].any());
+      assertNull(db[cn2].any());
+    },
+
+    testDebugDumpWithGraphNamedWithExamples: function () {
+      let vn1Before = [];
+      db[vn1].all().toArray().forEach(vertex => {
+        vn1Before.push(vertex._key);
+      });
+      vn1Before.sort();
+      let vn2Before = [];
+      db[vn2].all().toArray().forEach(vertex => {
+        vn2Before.push(vertex._key);
+      });
+      vn2Before.sort();
+      let cn1Before = [];
+      db[cn1].all().toArray().forEach(vertex => {
+        cn1Before.push(vertex._key);
+      });
+      cn1Before.sort();
+      let cn2Before = [];
+      db[cn2].all().toArray().forEach(vertex => {
+        cn2Before.push(vertex._key);
+      });
+      cn2Before.sort();
+      const graphBefore = db._graphs.document(graphName);
+      delete graphBefore._rev;
+      const fileName = fs.getTempFile() + "-debugDump";
+      const query = `FOR v,e, p IN 0..10 ANY  '${vn1}/value-1' GRAPH 'myGraph'  RETURN {vertex: p}`;
+      explainer.debugDump(fileName, query, {}, {examples: 10});
+      const outFileName = fs.getTempFile() + "-inspectDump";
+      explainer.inspectDump(fileName, outFileName);
+      recreateEmptyDatabase();
+      internal.load(outFileName);
+      const graphAfter = db._graphs.document(graphName);
+      delete graphAfter._rev;
+      assertEqual(graphBefore, graphAfter, "graph recreated after debugDump is not the same as the original");
+      let vn1After = [];
+      db[vn1].all().toArray().forEach(vertex => {
+        vn1After.push(vertex._key);
+      });
+      vn1After.sort();
+      let vn2After = [];
+      db[vn2].all().toArray().forEach(vertex => {
+        vn2After.push(vertex._key);
+      });
+      vn2After.sort();
+      let cn1After = [];
+      db[cn1].all().toArray().forEach(vertex => {
+        cn1After.push(vertex._key);
+      });
+      cn1After.sort();
+      let cn2After = [];
+      db[cn2].all().toArray().forEach(vertex => {
+        cn2After.push(vertex._key);
+      });
+      cn2After.sort();
+      assertEqual(vn1Before, vn1After);
+      assertEqual(vn2Before, vn2After);
+      assertEqual(cn1Before, cn1After);
+      assertEqual(cn2Before, cn2After);
+    },
+
+    testDebugDumpWithGraphNoNameInQuery: function () {
+      const collNameBefore = db[cn1].name();
+      const fileName = fs.getTempFile() + "-debugDump";
+      const query = `FOR v,e, p IN 0..10 ANY  '${vn1}/value-1' ${cn1} RETURN {from: e._from, to: e._to}`;
```

Comment:
Not sure if this works in cluster. 
Shouldn't the query start with `WITH ${vn1} ...`?